lib/matplotlib/axes/_base.py

```diff
@@ -734,6 +734,16 @@ def get_window_extent(self, *args, **kwargs):
         matplotlib.axis.Axis.get_tightbbox
         matplotlib.spines.Spine.get_window_extent
         """
+        kwargs.pop('renderer', None)
+        if len(args) > 1 or kwargs:
+            # when this expires coerce the API to
+            # get_window_extent(renderer=None) in line with
+            # Artist.get_window_extent
+            _api.warn_deprecated(
+                "3.6",
+                message="Passing any argument except 'renderer' to "
+                        "Axes.get_window_extent is deprecated in Matplotlib "
+                        "%(since)s and will be removed %(removal)s")
```

Comment:
I think(?) that's basically
```python
@_api.deprecated("3.6", "args")
@_api.deprecated("3.6", "kwargs")
def get_window_extend(self, *args, renderer=None, **kwargs): ...
```

Reply:
That would warn for `get_window_extent(renderer)` and `get_window_extent(renderer=r)`, which we should still support to be consistent with `Artist.get_window_extent`.

Reply:
Sorry, I think the correct invocation is
```python
@_api.delete_parameter("3.6", "args")
@_api.delete_parameter("3.6", "kwargs")
def gwe(self, renderer=None, *args, **kwargs): ...
```
?

Reply:
Oh, yes. Sure.